File: 哈希表/128. 最长连续序列/最长连续序列.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestConsecutive = function (nums) {
    // 这个题目的解题思路就是先把数组加到哈希表中去
    // 通过哈希表的查找,因为哈希表的查找是O(1)的复杂度
    let ansnumber = 0
    const numsset = new Set(nums)
    for (const number of numsset) {
        if (numsset.has(number - 1)) { // 如果有比这个数小一点的数就跳过
            continue
        }
        let nextnumber = number + 1
        while (numsset.has(nextnumber)) {
            nextnumber++
        }
        ansnumber = Math.max(ansnumber, nextnumber - number) // 更新值
    }
    return ansnumber
};

File: 哈希表/128. 最长连续序列/最长连续序列.cpp
#include<iostream>
#include<vector>
#include<unordered_set>
using namespace std;
// 时间复杂度O(n)，空间复杂度O(n)

class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> numsSet(nums.begin(), nums.end());
        int ansNumber = 0;
        for (int number : numsSet) {
            // 如果有比当前数小的数，跳过
            if (numsSet.find(number - 1) != numsSet.end())
                continue;
            int nextNumber = number + 1;
            while (numsSet.find(nextNumber) != numsSet.end())
                nextNumber++;
            ansNumber = max(ansNumber, nextNumber - number); // 更新最大长度
        }
        return ansNumber;
    }
};